src/McpHandler.php

```diff
@@ -108,7 +110,7 @@ public function listTools(): array
                         'additionalProperties' => false,
                     ],
                 ];
-            })->toArray(),
+            })->values()->toArray(),
```

Comment:
Fixes bug where the array keys are invalid when removing a tool from the collection.